src/compiler/checker.ts

```diff
@@ -21033,53 +21151,94 @@ namespace ts {
+                       }
+                       else if (node.kind === SyntaxKind.Parameter) {
+                           return grammarErrorOnNode(modifier, Diagnostics._0_modifier_cannot_appear_on_a_parameter, "override");
+                       }
+                       else if (node.parent.kind === SyntaxKind.ModuleBlock || node.parent.kind === SyntaxKind.SourceFile) {
+                           return grammarErrorOnNode(modifier, Diagnostics._0_modifier_cannot_appear_on_a_module_or_namespace_element, "override");
+                       }
+                       else if (node.kind !== SyntaxKind.MethodDeclaration &&
+                               node.kind !== SyntaxKind.PropertyDeclaration &&
+                               node.kind !== SyntaxKind.GetAccessor &&
+                               node.kind !== SyntaxKind.SetAccessor) {
+                               return grammarErrorOnNode(modifier, Diagnostics._0_modifier_can_only_appear_on_a_class_method_or_property_declaration, "override");
+                       }
+
+                       flags |= ModifierFlags.Override;
+                       break;
+
+                   case SyntaxKind.AsyncKeyword:
+                       if (flags & ModifierFlags.Async) {
+                           return grammarErrorOnNode(modifier, Diagnostics._0_modifier_already_seen, "async");
+                       }
+                       else if (flags & ModifierFlags.Ambient || isInAmbientContext(node.parent)) {
+                           return grammarErrorOnNode(modifier, Diagnostics._0_modifier_cannot_be_used_in_an_ambient_context, "async");
+                       }
+                       else if (node.kind === SyntaxKind.Parameter) {
+                           return grammarErrorOnNode(modifier, Diagnostics._0_modifier_cannot_appear_on_a_parameter, "async");
+                       }
+                       flags |= ModifierFlags.Async;
+                       lastAsync = modifier;
+                       break;
                 }
             }
 
             if (node.kind === SyntaxKind.Constructor) {
                 if (flags & ModifierFlags.Static) {
                     return grammarErrorOnNode(lastStatic, Diagnostics._0_modifier_cannot_appear_on_a_constructor_declaration, "static");
                 }
-                if (flags & ModifierFlags.Abstract) {
+                else if (flags & ModifierFlags.Abstract) {
                     return grammarErrorOnNode(lastStatic, Diagnostics._0_modifier_cannot_appear_on_a_constructor_declaration, "abstract");
                 }
                 else if (flags & ModifierFlags.Async) {
                     return grammarErrorOnNode(lastAsync, Diagnostics._0_modifier_cannot_appear_on_a_constructor_declaration, "async");
                 }
+                else if (flags & ModifierFlags.Override) {
+                    return grammarErrorOnNode(lastReadonly, Diagnostics._0_modifier_cannot_appear_on_a_constructor_declaration, "override");

```

Comment:
Should use `lastOverride`. And include a test for this error. Also for `override` of an index signature.

Reply:
Done